From 616dcf1b7535cf8e978fbe3e6cda228264cb7bba Mon Sep 17 00:00:00 2001
From: Erik Fong <quantumerikfong@gmail.com>
Date: Sun, 7 Dec 2025 20:22:07 -0500
Subject: [PATCH] Update `buffer_size()` to compile with libcamera 0.6.0

Still needs some testing, I was compiling it against arducams fork
---
 camera.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/camera.cpp b/camera.cpp
index 33ce548..e8b4adf 100644
--- a/camera.cpp
+++ b/camera.cpp
@@ -346,7 +346,7 @@ bool camera_create(const parameters_t *params, camera_frame_cb frame_cb,
     return true;
 }
 
-static int buffer_size(const std::vector<FrameBuffer::Plane> &planes) {
+static int buffer_size(Span<const FrameBuffer::Plane> planes) {
     int size = 0;
     for (const FrameBuffer::Plane &plane : planes) {
         size += plane.length;
